### ISSUE TITLE
Unnecessary instance call for `isinstance Iterable and isinstance SIzed`

### ISSUE THREAD

**Issue by tamuhey** [microsoft/pyright#1856]
```python
from typing import Any, Iterable, Sized

def f(v: Any) -> bool:
    return isinstance(v, Iterable) and isinstance(v, Sized)  # Unnecessary Instance call (actually not)

# but, for example, `list` is Iterable and Sized
l = []
assert isinstance(l, Iterable)
assert isinstance(l, Sized)

# and, `generator` is Iterable but not Sized
g = (x for x in range(10))
assert isinstance(g, Iterable)
assert not isinstance(g, Sized)
```

Outputs:

```
/tmp/foo.py
  /tmp/foo.py:5:40 - warning: Unnecessary isinstance call; "Iterable[Unknown]" is never an instance of "Sized" (reportUnnecessaryIsInstance)
0 errors, 1 warning, 0 infos
```

Pyright version: 1.1.138

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

This sample highlights another limitation in pyright that I didn't fix as part of this change because it is quite involved. It involves nested `isinstance` checks of protocol classes. The proper narrowed type in this case is an "intersection" of the two protocols. But there is no intersection type in the Python type system, so we don't currently support this. I've improved the handling so it doesn't narrow the type to `Never` in these circumstances.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.139, which I've just published. It will also be included in the next release of pylance.
